### ISSUE TITLE
Different query string behaviour introduced in 2.36.0

### ISSUE THREAD

**Issue by ldabiralai** [tulios/mappersmith#281]
Hi,

I'm experiencing an issue in 2.36.0 and later versions of mappersmith.

When passing an array as parameters they are not being sent as they were before.

2.36.0+ behaviour:
```js
// client with path of "something"
await client.fakeResource.fakeEndpoint({
  ids: ["a", "b"]
});

// request is made to /something with no query parameter
```

Previous (2.35.0) behaviour:
```js
// client with path of "something"
await client.fakeResource.fakeEndpoint({
  ids: ["a", "b"]
});

// request is made to /something?ids[]=a&ids[]=b
```

I've reproduced this here: https://codesandbox.io/s/magical-mclaren-2usjm?file=/src/App.js 
1. Open the url on the in page browser in a new tab
2. Inspect the network requests
3. Refresh and see that the call to `/something` has no query parameters added
4. Change the mappersmith dependency to version 2.35.0 on codesandbox
5. Refresh the page you opened previously and see the call to `/something` has the query parameters added

Is this intended?

Thanks!

**Comment by Yurickh** [comment]
Hello @ldabiralai! 👋

Indeed, I could create a breaking test that used to work for versions pre-2.35. 
This regression was unfortunately introduced by #245, but we're working on a patched version for 2.36 to get it fixed :)~

**Comment by ldabiralai** [comment]
Great, thanks @Yurickh!!

**Comment by klippx** [comment]
Thanks for a great bug report, a fix has been published in 2.36.4 and is also available in the brand new release 2.37.0
